packages/statusbar-extension/src/index.ts

```diff
@@ -46,8 +50,15 @@ export const STATUSBAR_PLUGIN_ID = '@jupyterlab/statusbar-extension:plugin';
 const statusBar: JupyterFrontEndPlugin<IStatusBar> = {
   id: STATUSBAR_PLUGIN_ID,
   provides: IStatusBar,
+  requires: [ISettingRegistry, IMainMenu, ICommandPalette],
   autoStart: true,
-  activate: (app: JupyterFrontEnd, labShell: ILabShell | null) => {
+  activate: (
+    app: JupyterFrontEnd,
+    settingRegistry: ISettingRegistry,
```

Comment:
Ever since #5845 we have been trying to make the core plugins less interdependent by making a lot of these tokens optional. Can you move `ISettingRegistry`, `IMainMenu`, and `ICommandPalette` to the optional list, and then check for them before using them in the `activate` function?